### PR TITLE
Memoize card mappings to grid

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

File: frontend/packages/console-shared/src/components/dashboard/DashboardGrid.tsx [hunk 1]
```diff
@@ -22,40 +22,53 @@ const mapCardsToGrid = (
...
-        <GridItem lg={6} md={6} sm={6}>
-          <Grid className="co-dashboard-grid">{mapCardsToGrid(mainCards, 'main')}</Grid>
-        </GridItem>
-        <GridItem key="right" lg={3} md={3} sm={3}>
-          <Grid className="co-dashboard-grid">{mapCardsToGrid(rightCards, 'right')}</Grid>
-        </GridItem>
-      </Grid>
-    );
+  const smallGrid =
+    !!containerRef.current && width !== null && width <= parseInt(breakpointLG.value, 10);
```
Comment (afreen23) [inline frontend/packages/console-shared/src/components/dashboard/DashboardGrid.tsx:28]: Just a minor suggestion:
May be we can use optional chaining here : https://github.com/openshift/console/blob/229af0a99a87218410a6f4704c89cbb5efa86646/frontend/public/components/utils/ref-width-hook.ts#L7-L10

Then ` width !== null `check can be removed.

Comment (rawagner) [reply frontend/packages/console-shared/src/components/dashboard/DashboardGrid.tsx]: fixed

**Pull request merged**